galleries/examples/lines_bars_and_markers/hat_graph.py

```diff
@@ -29,35 +29,32 @@ def hat_graph(ax, xlabels, values, group_labels):
         The group labels displayed in the legend.
     """
 
-    def label_bars(heights, rects):
-        """Attach a text label on top of each bar."""
-        for height, rect in zip(heights, rects):
-            ax.annotate(f'{height}',
-                        xy=(rect.get_x() + rect.get_width() / 2, height),
-                        xytext=(0, 4),  # 4 points vertical offset.
-                        textcoords='offset points',
-                        ha='center', va='bottom')
-
     values = np.asarray(values)
     x = np.arange(values.shape[1])
     ax.set_xticks(x, labels=xlabels)
-    spacing = 0.3  # spacing between hat groups
-    width = (1 - spacing) / values.shape[0]
     heights0 = values[0]
-    for i, (heights, group_label) in enumerate(zip(values, group_labels)):
-        style = {'fill': False} if i == 0 else {'edgecolor': 'black'}
-        rects = ax.bar(x - spacing/2 + i * width, heights - heights0,
-                       width, bottom=heights0, label=group_label, **style)
-        label_bars(heights, rects)
 
+    # Draw the hats
+    bars = ax.grouped_bar(
+        (values - heights0).T, bottom=heights0, tick_labels=xlabels,
+        labels=group_labels, edgecolor='black', group_spacing=0.8)
+
+    # Attach a text label on top of each bar
+    for bc, heights in zip(bars.bar_containers, values):
+        ax.bar_label(bc, heights, padding=4)
+
+    # Remove the color that would otherwise show in the first group's legend entry
+    bars.bar_containers[0].patches[0].set_facecolor('none')
```

Comment:
Better do as `grouped_bar(..., color=['none', 'C1'])`?

Reply:
I actually did that first, but then noticed the docstring assumes an arbitrary number of groups (M).  The linked paper only uses two groups.  I didn’t find many examples with an image search, but [this](https://www.ifourtechnolab.com/power-bi-templates/hat-graph) does show more groups.

Reply:
Fair point. Since the colors don’t need to be the exact length, you could do `color=[„none“] + all_colors_in_the_color_cycle`. That would also make the first color C0, which may be more appropriate.